From a5d5d8578176fb637353b3cc3101c221e63b0bc0 Mon Sep 17 00:00:00 2001
From: Sriram R <quic_srirrama@quicinc.com>
Date: Tue, 7 Nov 2023 16:56:07 +0530
Subject: [PATCH] hostapd: avoid m2/m4 validation

Some clients doesnt send link KDEs properly. Until
its resolved, avoid these validations.

Signed-off-by: Sriram R <quic_srirrama@quicinc.com>
---
 src/ap/wpa_auth.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ap/wpa_auth.c b/src/ap/wpa_auth.c
index 13be8f1..46555c1 100644
--- a/src/ap/wpa_auth.c
+++ b/src/ap/wpa_auth.c
@@ -3304,6 +3304,8 @@ static int wpa_auth_validate_ml_kdes_m2(struct wpa_state_machine *sm,
 	int i;
 	unsigned int n_links = 0;
 
+	return 0;
+
 	if (sm->mld_assoc_link_id < 0)
 		return 0;
 
@@ -4546,6 +4548,8 @@ static int wpa_auth_validate_ml_kdes_m4(struct wpa_state_machine *sm)
 	u16 key_data_length;
 	size_t mic_len;
 
+	return 0;
+
 	if (sm->mld_assoc_link_id < 0)
 		return 0;
 
-- 
2.17.1

